### PR TITLE
Everything i know about ssd

### LINKED ISSUE #74: （我所知道关于 SSD 的知识）
@FreeCodeCamp-Chengdu 我添加了一篇想翻译的新文章：

https://kcall.co.uk/ssd/index.html

### PR DESCRIPTION
**Pull request by luojiyin1987**
[PR-82](https://github.dev/FreeCodeCamp-Chengdu/Wiki/pull/82) [PR-82](https://codespaces.new/FreeCodeCamp-Chengdu/Wiki/pull/82) [PR-82](https://gitpod.io/?autostart=true#https://github.com/FreeCodeCamp-Chengdu/Wiki/pull/82) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=FreeCodeCamp-Chengdu&utm_campaign=badge_info)

@FreeCodeCamp-Chengdu I create a Pull Request to close #74

### CODE REVIEW

**Comment by TechQuery** [comment]
@luojiyin1987 处理冲突

**Comment by luojiyin1987** [comment]
@TechQuery  已经解决冲突

File: _posts/Article/Translation/everything-I-know-about-ssd.md [hunk 1]
```diff
@@ -4,12 +4,273 @@ authorURL: ""
...
+
+读取或编程操作涉及芯片控制器使用块解码器选择相关块，然后使用页解码器选择块中的页。芯片控制器还负责激活正确的模拟电路，以产生编程和擦除操作所需的电压。
+
+虽然每行的单元格数名义上与页面大小相等，但每行的实际单元格数却高于每页的标称容量。这是因为每个页面都包含一组备用单元和数据单元。备用单元存储该页面的 ECC 位以及页面的物理地址到逻辑地址映射。控制器还可在备用区中保存有关页面的其他元数据信息。在读取操作过程中，整个页面（包括备用区中的位）都会传输到控制器。控制器中的 ECC 逻辑会检查并纠正读取的数据。在编程操作期间，控制器会将用户数据和 ECC 位传送到闪存。
+
+系统启动时，控制器会扫描整个闪存阵列中每个页面的空闲区域，将逻辑地址到物理地址的映射加载到自己的内存中（控制器中可能有其他保存映射数据的技术）。控制器在闪存转换层（FTL）中保存逻辑地址到物理地址的映射。FTL 还执行垃圾回收，清除写入后的无效页面，并执行损耗均衡，确保所有闪存块都得到均衡使用。
+
+由于闪存不支持原地更新，所以在内容可以被编程之前，页面需要被擦除；但不同于以页面粒度进行的编程或读取操作，擦除操作是以块粒度执行的。
+
+## 2D 和 3D, 和 Layers（层）
```
Comment (TechQuery) [inline _posts/Article/Translation/everything-I-know-about-ssd.md:62]:
```suggestion
## 2D、 3D 和 Layers（层）
```

File: _posts/Article/Translation/everything-I-know-about-ssd.md [hunk 2]
```diff
@@ -4,12 +4,273 @@ authorURL: ""
...
+
+在闪存架构中，一个平面闪存块，也就是一个二维的存储单元阵列，被称为 2D 闪存。如果一个或多个这样的阵列叠放在一起，那么就是 3D 闪存。3D NAND 闪存建立在一块芯片上，最多可达 32 层，是为了在平面闪存达到尺寸限制时降低成本而设计的：3D 闪存的生产成本几乎与 2D 相当，但存储容量大大增加。在 2D 和 3D 中，每个页面（即行）中的存储单元由单词线连接，而页面中的每个偏移量（即列）中的存储单元则通过位线连接（简单地说）。
+
+3D 闪存与层叠式闪存不同，后者是将多个独立的超薄芯片堆叠在一起。这种做法成本高得令人望而却步。大多数现代消费级固态硬盘（2010 年代）都使用 3D TLC 闪存。
+
+## 我能看到其中一个单元吗？
+
+终端用户闪存的单元尺寸极小，通常为 15 纳米，范围从 43 纳米到 12 纳米不等。实际上，单元尺寸或单元直径具有误导性，因为所述尺寸并非单元任何尺寸的测量值，而是芯片上离散组件之间距离的度量。芯片上的硅层厚度约为 0.5 至 3 纳米：相比之下，氢原子的直径为 0.1 纳米，而芯片制造中使用的硅原子为 0.2 纳米。一纳米 (nm) 确实非常小，仅为一米的十亿分之一，打个比方，如果一毫米相当于一颗标准弹珠的大小（约 13 毫米），那么一米就相当于地球的大小。十亿倍的威力令人印象深刻。
+
+## SLC, MLC, TLC, QLC 及其未来发展
```
Comment (TechQuery) [inline _posts/Article/Translation/everything-I-know-about-ssd.md:72]:
```suggestion
## SLC、MLC、TLC、QLC 及其未来发展
```

File: _posts/Article/Translation/everything-I-know-about-ssd.md [hunk 3]
```diff
@@ -4,12 +4,273 @@ authorURL: ""
...
+
+不幸的是，当双级单元被开发出来时，它被称为多级单元，并被赋予了 MLC 这个缩写，这迫使每个人在想要指代多个级别的存储单元时都要费力地打出“多级单元”这个词。如果它被称为双级单元，我们就可以自由地使用 DLC、TLC 和 QLC，并用 MLC 来描述所有这些，但现在为时已晚。如果闪存只停留在 SLC 阶段，有其是/否、一/零的状态，这些解释将更容易写，希望也更容易理解。
+
+在多级单元中，物理 NAND 页面代表两个或更多的逻辑页面。属于 MLC 的两个位分别映射到两个逻辑页面。奇数编号的页面（包括零）映射到最不重要的（右手）位，偶数编号的页面映射到最重要的（左手）位。同样，属于 TLC 的三个位分别映射到三个逻辑页面，而 QLC 映射到四个逻辑页面（TLC 和 QLC 的页面编号是未知的）。
+
+多层单元需要支持的位数越多，其性能就越受影响。对于 SLC，控制器只需要检查是否超过了一个阈值。而 MLC 单元可以有四个值，TLC 有八个，QLC 有 16 个。为了读取单元的正确值，SSD 控制器需要使用精确的电压和多次读取来确定单元中的电荷。很明显，如果一个物理页面支持多个逻辑页面，那么该页面的读写频率将高于 SLC 页面，从而影响其使用寿命。此外，显而易见的是，TLC SSD 只需要 SLC 设备三分之一的物理单元，因此我的 120 GB TLC SSD 实际上只包含 40 GB 的 NAND 单元。
+
+高使用率的企业级 SSD 曾经是 SLC 的天下，因为它具有更高的速度、耐用性、可靠性和读/写能力，而 MLC 和 TLC 也逐渐被企业级应用所接受。终端消费级 SSD 市场则获得了更便宜、容量更大，但速度更慢、更脆弱的多层单元。
+
+**为什么空代表一？**
```
Comment (TechQuery) [inline _posts/Article/Translation/everything-I-know-about-ssd.md:84]:
```suggestion
### 为什么空代表一？
```

File: _posts/Article/Translation/everything-I-know-about-ssd.md [hunk 4]
```diff
@@ -4,12 +4,273 @@ authorURL: ""
...
+
+## 垃圾回收难题
+
+垃圾回收可以在后台进行（当主机空闲时），也可以在前台进行（当写入需要时）。虽然后台垃圾回收看起来更可取，但它也有缺点。如果主机在空闲时使用节能模式，垃圾回收要么会等待设备重新启动，从而导致用户延迟以完成垃圾回收，要么会唤醒设备并缩短电池寿命，而此时主机处于“空闲”状态。此外，垃圾回收并不知道它正在回收的数据。不可避免的是，一些数据在进行垃圾回收后不久就会被删除，从而导致另一轮垃圾回收和随之而来的额外和不必要的写入（写入放大，即实际写入与数据写入的比率）。前台垃圾回收看似与性能背道而驰，但它避免了节能问题，只在实际需要时才进行写入，并且借助快速缓存和高度发达的垃圾回收算法，不会给用户带来明显的性能损失。现代垃圾回收的趋势似乎是前台回收，或者前台和后台回收的组合。
+
+基于前台垃圾回收，以及大多数用户活动都是随机的，那么不可避免的结论是，SSD 将在其大部分生命周期中处于满容量状态，如果我们指的是可用块，即使分配给主机的空间看起来很低。
+
+然而，SSD 还存在另一个潜在问题，这与一个历史事件有关：文件系统的设计方式。
+
+## 文件系统 - 非眼见为实
```
Comment (TechQuery) [inline _posts/Article/Translation/everything-I-know-about-ssd.md:170]:
```suggestion
## 文件系统 —— 非眼见为实
```

**Review by TechQuery — CHANGES_REQUESTED** [review]

**Review by TechQuery — APPROVED** [review]

**Pull request merged**